calico-vpp-agent/services/service_server.go

```diff
@@ -81,6 +89,64 @@ func (s *Server) SetOurBGPSpec(nodeBGPSpec *common.LocalNodeSpec) {
 	s.nodeBGPSpec = nodeBGPSpec
 }
 
+func (s *Server) ParseServiceAnnotations(annotations map[string]string) *serviceInfo {
+	var err []error
+	svc := &serviceInfo{}
+	for key, value := range annotations {
+		switch key {
+		case LBTypeAnnotation:
+			switch value {
+			case "ECMP":
+				svc.lbType = lbTypeECMP
+			case "Maglev":
+				svc.lbType = lbTypeMaglev
+			case "MaglevDSR":
+				svc.lbType = lbTypeMaglevDSR
+			default:
+				svc.lbType = lbTypeECMP // default value
+				err = append(err, errors.Errorf("Unknown value %s for key %s", value, key))
+			}
+		case HashConfigAnnotation:
+			hashConfigList := strings.Split(strings.TrimSpace(value), ",")
+			for _, hc := range hashConfigList {
+				switch strings.TrimSpace(hc) {
+				case "srcport":
+					svc.hashConfig = append(svc.hashConfig, hashConfigSrcport)
+				case "dstport":
+					svc.hashConfig = append(svc.hashConfig, hashConfigDstport)
+				case "srcaddr":
+					svc.hashConfig = append(svc.hashConfig, hashConfigSrcaddr)
+				case "dstaddr":
+					svc.hashConfig = append(svc.hashConfig, hashConfigDstaddr)
+				case "iproto":
+					svc.hashConfig = append(svc.hashConfig, hashConfigIproto)
+				case "reverse":
+					svc.hashConfig = append(svc.hashConfig, hashConfigReverse)
+				case "symmetric":
+					svc.hashConfig = append(svc.hashConfig, hashConfigSymmetric)
+				default:
+					err = append(err, errors.Errorf("Unknown value %s for key %s", value, key))
+				}
+			}
+		case KeepOriginalPacketAnnotation:
+			switch value {
+			case "true":
+				svc.keepOriginalPacket = true
+			case "false":
+				svc.keepOriginalPacket = false
+			default:
+				err = append(err, errors.Errorf("Unknown value %s for key %s", value, key))
+			}
+		default:
+			continue
+		}
+		if len(err) != 0 {
+			s.log.Warnf("Error parsing: %s", err)
```

Comment:
This should probably be logged as an error.
Also we should provide context info (i.e. which is the service name it corresponds to, which is the annotation that triggered the error, etc..